Test for 1-my_list.py


case 0: import
>>> MyList = __import__('1-my_list').MyList
>>>


case 1: common case

>>> my_list = MyList()
>>> my_list.append(10)
>>> my_list.append(20)
>>> my_list.append(30)
>>> my_list.append(400)
>>> my_list.append(500)
>>> print(my_list)
[10, 20, 30, 400, 500]
>>> my_list.print_sorted()
[10, 20, 30, 400, 500]
>>> print(my_list)
[10, 20, 30, 400, 500]
>>>


case 2: not defined object

>>> MyList = __import__('1-my_list').MyList
>>> my_list2.print_sorted()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'my_list2' is not defined
>>>


case 3: passed a list

>>> x = [10, 88, 5]
>>> my_list = MyList(x)
>>> my_list.print_sorted()
[10, 88, 5]
>>>


case 4: empty list

>>> x = []
>>> my_list = MyList(x)
>>> my_list.print_sorted()
[]
>>>


case 5: unique number

>>> my_list = MyList()
>>> my_list.append(1)
>>> my_list.print_sorted()
[1]
>>> print(my_list)
[1]
>>>


case 6: same number

>>> my_list2 = MyList()
>>> my_list2.append(2)
>>> my_list2.append(2)
>>> my_list2.append(2)
>>> my_list2.append(2)
>>> my_list2.print_sorted()
[2, 2, 2, 2]
>>>


case 7: negative numbers

>>> MyList = __import__('1-my_list').MyList
>>> my_list = MyList()
>>> my_list.append(8)
>>> my_list.append(3)
>>> my_list.append(-15)
>>> my_list.append(90)
>>> my_list.append(-105)
>>> my_list.print_sorted()
[8, 3, -15, 90, -105]
>>>
